command/sub8_missions/sub8/tx_sub.py

```diff
@@ -33,14 +33,15 @@ class _Sub(object):
     def __init__(self, node_handle):
         self._node_handle = node_handle
 
+    @util.cancellableInlineCallbacks
     def _init(self):
-        self._moveto_action_client = action.ActionClient(self._node_handle, 'moveto', MoveToAction)
-        self._odom_sub = self._node_handle.subscribe('odom', Odometry)
-        self._trajectory_sub = self._node_handle.subscribe('trajectory', PoseTwistStamped)
-        self._trajectory_pub = self._node_handle.advertise('trajectory', PoseTwistStamped)
+        self._moveto_action_client = yield action.ActionClient(self._node_handle, 'moveto', MoveToAction)
+        self._odom_sub = yield self._node_handle.subscribe('odom', Odometry)
+        self._trajectory_sub = yield self._node_handle.subscribe('trajectory', PoseTwistStamped)
+        self._trajectory_pub = yield self._node_handle.advertise('trajectory', PoseTwistStamped)
```

Comment:
FYI, yielding on these isn't required. It doesn't hurt, of course. It's kind of annoying that there's not some good way to know/indicate what returns a Deferred and what doesn't. You could postfix functions/methods that return a Deferred with "_df" or something, but that's ugly.